src/pages/instances/EditInstanceForm.tsx

```diff
@@ -112,11 +116,22 @@ const EditInstanceForm: FC<Props> = ({ instance }) => {
     },
   });
 
-  const updateSection = (newItem: string) => {
-    if (section === YAML_CONFIGURATION && newItem !== YAML_CONFIGURATION) {
+  const updateSection = (newSection: string) => {
+    if (
+      activeSection === slugify(YAML_CONFIGURATION) &&
+      newSection !== slugify(YAML_CONFIGURATION)
+    ) {
       void formik.setFieldValue("yaml", undefined);
     }
-    setSection(newItem);
+    if (newSection === slugify(INSTANCE_DETAILS)) {
+      navigate(
+        `/ui/${project}/instances/detail/${instance.name}/configuration`
+      );
+    } else {
+      navigate(
+        `/ui/${project}/instances/detail/${instance.name}/configuration/${newSection}`
```

Comment:
We seem to call `slugify` in just so many places. We can remove all the calls and instead call it once here on navigation. Then add a function for sluggish comparison, to fix the matching checks:
`const isSlugMatch = (a: string, b: string) => return  slugify(a) === slugify(b)`

Reply:
This is causing a lot of unnecessary calls to `slugify` because one of the two strings will be already slugified, and we still call `slugify` on it in the `isSlugMatch` function.
I'd rather keep it as it is, it looks much cleaner as we can understand clearly when a label gets slugified. Also `activeSection` can be undefined, forcing you to take that into account as well - either in the `isSlugMatch` function, or upstream.